drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -76,14 +76,14 @@ static int pd_intel_adsp_init(const struct device *dev)
 
 #define DT_DRV_COMPAT intel_adsp_power_domain
 
-#define POWER_DOMAIN_DEVICE(id)							\
-	static struct pg_bits pd_pg_reg##id = {					\
-		.SPA_bit = DT_INST_PROP(id, bit_position),			\
-		.CPA_bit = DT_INST_PROP(id, bit_position),			\
-	};									\
-	PM_DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_pm_action);			\
-	DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_init, PM_DEVICE_DT_INST_GET(id),\
-			      &pd_pg_reg##id, NULL, POST_KERNEL,                \
+#define POWER_DOMAIN_DEVICE(id)								\
+	static struct pg_bits pd_pg_reg##id = {						\
+		.SPA_bit = DT_INST_PROP(id, bit_position),				\
+		.CPA_bit = DT_INST_PROP(id, bit_position),				\
+	};										\
+	PM_DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_pm_action);				\
+	DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_init, PM_DEVICE_DT_INST_GET(id),	\
+			      &pd_pg_reg##id, NULL, CONFIG_POWER_DOMAIN_INIT_PRIORITY,	\
 			      CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, NULL);
```

Comment:
I think you swapped the wrong argument here, `POST_KERNEL` became `CONFIG_POWER_DOMAIN_INIT_PRIORITY`, guess you wanted to change `CONFIG_KERNEL_INIT_PRIORITY_DEFAULT` instead.